src/lists/vector.js

```diff
@@ -0,0 +1,74 @@
+const Node = require('../base')
+const iavector = require('iavector')
+const Block = require('@ipld/block')
+
+const _type = 'IPLD/Experimental/Vector/0'
+
+class Vector extends Node {
+  get _type () {
+    return _type
+  }
+
+  get (args) {
+    let path = args.path.split('/').filter(Boolean)
+    let index = parseInt(path.shift(), 10)
+
+    if (!Number.isInteger(index)) {
+      throw new Error('Vector type can only look up integers ' + index + ', ' + args)
+    }
+
+    let traversal = iavector.traverseGetOne(this.data, index)
+
+    if (!traversal) {
+      throw new Error(`Vector does not contain index`)
+    }
+
+    if (traversal.value) {
+      if (!path.length) {
+        return { result: traversal.value }
+      }
+      // user wants to go deeper into returned object
+      let info = { method: 'get', args: { path: path.join('/') } }
+      let call = { info, target: traversal.value, proxy: true }
+      return { call }
+    }
+
+    path = `${traversal.nextIndex}/${path.join('/')}`
+    return { call: { target: traversal.nextId, info: { method: 'get', args: { path } }, proxy: true } }
+  }
+}
+
+Vector._type = _type
+
+async function * runConstruction (construction, codec) {
+  while (true) {
+    let c = 0
```

Comment:
`c` is used like a boolean. Would it make sense to change it to `let isConstructed = false`?

Reply:
yes, a leftover from experimenting with more complex constructions (see below in the file), it just gets too messy. will make it a boolean but this whole thing needs to be revisited because the write interfaces need to be more crisp -- this is driving in the same direction as the WASM versions of these that you're thinking about - operations that are one step at a time and have some form of state management. a "construction" in this sense is a mutation but the state management becomes so messy.